EZSwiftExtensionsTests/ArrayTests.swift

```diff
@@ -247,4 +247,19 @@ class ArrayTests: XCTestCase {
         XCTAssertEqual(duplicatedElementsArrayOrderMaintained.unique(), [2, 1, 3])
         
     }
+    
+    func testElementForIndexPath() {
+        let array = [["a", "b", "c"],["1", "2"]]
```

Comment:
So what does this array represent? An M * N matrix (in your test case its 3 * 2). If so is it a matrix, and I presume section is the row number and column number, how is it represented in this array. 

Traditionally you represent it via a either a row major or a column major format https://en.wikipedia.org/wiki/Row-_and_column-major_order. Your representation is neither of those. 

The other possibility is its a cartesian product. In which case this test case is wrong.

In either case I would not vouch for this extension. From a Data structure standpoint, its better design to present a matrix struct with checks and balances to prevent data from being misrepresented (for instance utilizing this method on a 3 dimensional matrix would not make sense).